test/tests/functional/pbs_acct_log_trunc.py

```diff
@@ -0,0 +1,147 @@
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestAcctlogTrunc(TestFunctional):
+    """
+    Test that the resources_used information doesn't get truncated
+    in accounting log records
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        a = {'type': 'string', 'flag': 'h'}
+        self.server.manager(MGR_CMD_CREATE, RSC, a, id='foo_str')
+
+    def test_long_resource_end(self):
+        """
+        Test to see if a very long string resource is neither truncated
+        in the job's resources_used attr or the accounting log at job end
+        """
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'job_history_enable': 'True'})
+
+        # Create a very long string - the truncation was 2048 characters
+        # 4096 is plenty big to show it
+        hstr = ""
+        for i in range(4096):
+            hstr += "1"
+
+        hook_body = "import pbs\n"
+        hook_body += "e = pbs.event()\n"
+        hook_body += "hstr=\'" + hstr + "\'\n"
+        hook_body += "e.job.resources_used[\"foo_str\"] = hstr\n"
+
+        a = {'event': 'execjob_epilogue', 'enabled': 'True'}
+        self.server.create_import_hook("ep", a, hook_body)
+
+        J = Job()
+        J.set_attributes({ATTR_m: 'e'})
+        J.set_sleep_time(1)
+        jid = self.server.submit(J)
+
+        # Make sure the resources_used value hasn't been truncated
+        self.server.expect(JOB, {'job_state': 'F'}, id=jid, extend='x')
+        self.server.expect(
+            JOB, {'resources_used.foo_str': hstr}, extend='x', max_attempts=1)
+
+        # Make sure the accounting log hasn't been truncated
+        log_match = 'resources_used.foo_str=' + hstr
+        m = self.server.accounting_match(
+            "E;%s;.*%s.*" % (jid, log_match), regexp=True)
+        self.assertTrue(m)
+
+        # Make sure the server log hasn't been truncated
+        log_match = 'resources_used.foo_str=' + hstr
+        m = self.server.log_match("%s;.*%s.*" % (jid, log_match), regexp=True)
+        self.assertTrue(m)
+
+        # Make sure emails are not truncated
+        mailfile = os.environ['MAIL']
+        if not os.path.isfile(mailfile):
+            self.logger.info("Mail file do not exist or mail not setup.\
```

Comment:
Can you change this message to "Mail file does not exist or mail is not setup." ?